src/utils/sets.ts

```diff
@@ -32,3 +34,14 @@ export function setHasDiff<T>(a: Set<T>, b: Set<T>): boolean {
         return true; // different lengths means they are naturally diverged
     }
 }
+
+/**
+ * Determines the values added and removed between two sets.
+ * @param a The first set. Must be defined.
+ * @param b The second set. Must be defined.
+ * @returns The difference between the values in each set.
+ */
+export function setDiff<T>(a: Set<T>, b: Set<T>): Diff<T> {
+    const keyDiff = arrayDiff(Array.from(a), Array.from(b));
+    return { added: keyDiff.added, removed: keyDiff.removed };
+}
```

Comment:
if the return type of `arrayDiff` gets changed, this becomes:
```suggestion
export function setDiff<T>(a: Set<T>, b: Set<T>): Diff<T> {
    return arrayDiff(Array.from(a), Array.from(b));
}
```